m2k/serverless-workflow-m2k/src/main/resources/m2k.sw.yml

```
@@ -0,0 +1,226 @@
+    actions:
+      - name: printSystemOut
+        functionRef:
+          refName: systemOut
+          arguments:
+            message: '${"m2k workflow: " + $WORKFLOW.instanceId + " transformation ID: " + .transformId }'
+    transition: SendEmail
+  - name: SendEmail
+    type: operation
+    actions:
+    - functionRef:
+        refName: sendEmail
+        arguments:
+          inbox_id: $SECRET.mailtrap_inbox_id | tonumber
+          to:
+            - email: .user
+              name: "user"
+          from:
+            email: $SECRET.sender_email
+            name: "Move2Kube"
+          subject: " \"Transformation \" + .transformationId "
+          html: '"Please go to your <a href=\"" + $SECRET.move2kube_url + "/workspaces/"+ .workspaceId + "/projects/" + .projectId + "/outputs/"+ .transformId + "\">Move2Kube instance</a> and answers to questions in order to continue the Move2Kube workflow"'
+    transition: SaveTransformationOutput
+  - name: SaveTransformationOutput
+    type: operation
+    actions:
+      - functionRef:
+          refName: sendCloudEvent
+          arguments:
+            HEADER_Ce-Id: "5678"
+            HEADER_Ce-Specversion: "1.0"
+            HEADER_Ce-Type: "save-transformation"
+            HEADER_Ce-Source: "m2k_swf"
+            HEADER_Content-Type: "application/json"
+            githubRepo: .repo
+            branch: .targetBranch
+            token: .token
+            workspaceId: .workspaceId
+            projectId: .projectId
+            transformId: .transformId
+            workflowCallerId: $WORKFLOW.instanceId
+    transition: WaitForSaveTransformationCompletion
+  - name: WaitForSaveTransformationCompletion
+    type: switch
+    eventConditions:
+      - eventRef: transformationSavedEvent
+        transition: TransformationSaved
+      - eventRef: errorEvent
+        transition: TransformationError
+    defaultCondition:
+      transition: TransformationError
+    timeouts:
+      eventTimeout: PT3M
```

Maybe you can also intercept the timeout error (newly introduced in 2.0.0-SNAPSHOT) and react appropriately? Otherwise it will follow the default.
See https://github.com/parodos-dev/serverless-workflow-examples/blob/b6cda642a367983346e0ccf912b197dd5effed8f/escalation-eda/escalation-swf/src/main/resources/ticketEscalation.sw.yaml#L15 and https://github.com/parodos-dev/serverless-workflow-examples/blob/b6cda642a367983346e0ccf912b197dd5effed8f/escalation-eda/escalation-swf/src/main/resources/ticketEscalation.sw.yaml#L76

---

(of course this can be a future enhancement, but it'd be great if we all adopt the same patterns to prove consistent implementations to the readers)

---

BTW, In real life, the timeout will be much higher, months at least
I'll check to use the latest version with this feature but last time I tried I had some issues running the workflow, thus the needed revision, but I'll try again, can't hurt :)

---

Not sure this is needed to do this timeout error management here, per the example https://sonataflow.org/serverlessworkflow/latest/use-cases/timeout-showcase-example.html, if the timeout expire, the switch will go to the default condition, I'll add `TransformationTimedOut` to make that clear

---

if you mean to transition by default to `TransformationTimedOut`, it might be misleading as all the uncaught errors would follow the same default.
so I agree that this can be postponed so I'll add one more point to the final comment below for this reason. you can resolve this comment